### PR TITLE
Add io-functions-cgn-operator-search

### PR DESCRIPTION
**Pull request by ElfoLiNk**
https://github.com/pagopa/io-functions-cgn-operator-search

### CODE REVIEW

**Review by pasqualedevita — CHANGES_REQUESTED** [review]
thanks @ElfoLiNk can u move this pipeline into cgn-onboarding-portal-projects?

File: azure-devops/projects/io-backend-projects/io-functions-cgn-operator-search.tf [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    repo_id               = "${var.io-functions-cgn-operator-search.repository.organization}/${var.io-functions-cgn-operator-search.repository.name}"
+    branch_name           = var.io-functions-cgn-operator-search.repository.branch_name
+    yml_path              = "${var.io-functions-cgn-operator-search.repository.pipelines_path}/code-review-pipelines.yml"
+    service_connection_id = azuredevops_serviceendpoint_github.io-azure-devops-github-pr.id
+  }
+
+  variable {
+    name         = "DANGER_GITHUB_API_TOKEN"
+    secret_value = data.azurerm_key_vault_secret.key_vault_secret["DANGER-GITHUB-API-TOKEN"].value
+    is_secret    = true
```
Comment (pasqualedevita) [inline azure-devops/projects/io-backend-projects/io-functions-cgn-operator-search.tf:58]:
```suggestion
    is_secret    = true
    allow_override = false
```

File: azure-devops/projects/io-backend-projects/io-functions-cgn-operator-search.tf [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+# Allow deploy pipeline to access Azure PROD-IO subscription service connection, needed to interact with Azure resources
+resource "azuredevops_resource_authorization" "io-functions-cgn-operator-search-deploy-azurerm-PROD-IO-auth" {
+  depends_on = [azuredevops_serviceendpoint_azurerm.PROD-IO, azuredevops_build_definition.io-functions-cgn-operator-search-deploy, time_sleep.wait]
+
+  project_id    = azuredevops_project.project.id
+  resource_id   = azuredevops_serviceendpoint_azurerm.PROD-IO.id
+  definition_id = azuredevops_build_definition.io-functions-cgn-operator-search-deploy.id
+  authorized    = true
+  type          = "endpoint"
+}
```
Comment (pasqualedevita) [inline azure-devops/projects/io-backend-projects/io-functions-cgn-operator-search.tf:176]: PROD-Esercenti

File: azure-devops/projects/io-backend-projects/io-functions-cgn-operator-search.tf [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+# Allow deploy pipeline to access NPM service connection, needed to publish sdk packages to the public registry
+resource "azuredevops_resource_authorization" "io-functions-cgn-operator-search-deploy-npm-auth" {
+  depends_on = [azuredevops_serviceendpoint_npm.pagopa-npm-bot, azuredevops_build_definition.io-functions-cgn-operator-search-deploy, time_sleep.wait]
+
+  project_id    = azuredevops_project.project.id
+  resource_id   = azuredevops_serviceendpoint_npm.pagopa-npm-bot.id
+  definition_id = azuredevops_build_definition.io-functions-cgn-operator-search-deploy.id
+  authorized    = true
+  type          = "endpoint"
+}
```
Comment (pasqualedevita) [inline azure-devops/projects/io-backend-projects/io-functions-cgn-operator-search.tf:187]: remove npm connection

**Review by pasqualedevita — CHANGES_REQUESTED** [review]

**Comment by ElfoLiNk** [comment]
@pasqualedevita do we need other changes?

**Review by pasqualedevita — APPROVED** [review]
LGTM but only for code review.
For deploy I think we need some changes to manage uat and prod environments

**Pull request merged**